engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/JobExecutionEngine.scala

```diff
@@ -50,12 +50,13 @@ final class JobExecutionEngine(
     context.newCachedThreadPool(
       "prioritized-job-pool",
       2,
-      Integer.MAX_VALUE,
+      4,
+      20,
       false
     )
 
   private val backgroundJobExecutor: ExecutorService =
-    context.newFixedThreadPool(1, "background-job-pool", false)
+    context.newCachedThreadPool("background-job-pool", 1, 4, 20, false)
```

Comment:
Deserialization jobs are quite expensive in terms of execution time and there is no need to execute them sequentially. 
This is particularly important when modules' invalidation is triggered.

Reply:
> Deserialization jobs are quite expensive in terms of execution time

Are they? We are supposed to have _lazy deserialization_ - in such case most of the deserialization is happening later, during code execution, anyway. However I can see that `SuggestionsCache` is still relying on old `java.io.ObjectInputStream` mechanism - probably because we are not benchmarking `SuggestionsCache` in our [startup benchmarks](...)